2-WebApp-graph-user/2-5-HybridFlow/README.md

```diff
@@ -0,0 +1,480 @@
+</details>
+
+## About the code
+
+<details>
+ <summary>Expand to see more information about the code</summary>
+
+## Server setup
+
+The entire application is built using the [Microsoft Identity Web](https://docs.microsoft.com/en-us/azure/active-directory/develop/microsoft-identity-web) library and [ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/introduction-to-aspnet-core?view=aspnetcore-6.0) using [Razor](https://docs.microsoft.com/en-us/aspnet/web-pages/overview/getting-started/introducing-razor-syntax-c) pages.
+
+The main entry point of this program is found in the `Program.cs` file. With **ASP.NET Core 6.0** [the Startup.cs file has been merged with the Program.cs file](https://docs.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-6.0?view=aspnetcore-6.0#web-app-template-improvements). As a result, much of the configuration you'd see in older samples in the `Startup.cs` file now take place in the `Program.cs` file. Also, because this is based on C# 9 [the main method no longer needs to be explicitly stated and a top-level statement is used instead.](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/program-structure/top-level-statements)
+
+In the `Program.cs` file you will see the [WebApplicationBuilder](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.webapplicationbuilder?view=aspnetcore-6.0) **builder** which injects all dependencies into your application.
+    * The authentication dependencies are some of the first to be injected into the application using the `AddAuthentication` method. The `OpenIdConnectDefaults.AuthenticationScheme` is provided to configure the application to use the [OpenID Connect protocol](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc).
+    * The `AddMicrosoftIdentityWebApp` called to further configure the application to use the authentication services required by **Microsoft Identity**. Because this is using a hybrid flow the `OnAuthorizationCodeReceived` event is customized to allow for the server to request an **authorization code** to be retrieved from the server and sent to the client to be redeemed while also acquiring a token to be stored server side. Both the server-side **authentication token** and client-side **authorization code** are redeemed using the `ConfidentialClientService` which is discussed later. Because this application is configured to use [PKCE](https://datatracker.ietf.org/doc/html/rfc7636) the `code_challenge` value is extracted from the URL that would have been used by the flow automatically and is passed to the `ConfidentialClientService` to acquire the needed **authentication token** and **authorization code**. The **authorization code** is then stored in the **session** to be placed in a JavaScript variable rendered server-side to be redeemed for an **authentication token**. This is discussed in more detail later.
```

Comment:
>Because this is using a hybrid flow the `OnAuthorizationCodeReceived` event is customized to [](http://example.com/codeflow?start=154&length=92)

nit: is customized as follows ..... then break the remaining text into bullet points